### PR TITLE
Fix Recognize, Passwords, Cookie, Text Editor, Search, and deleting mailbox FPs

### PR DESCRIPTION
**Pull request by EsadCetiner**
This PR fixes the following
1. Adds support for Nextcloud Recognize
2. Fix FP when changing passwords
3. Fix FP when changing sorting direction in Files
4. Fix FP when deleting a mailbox
5. Fix FP with cookies
6. Fix FP with text editor
7. Fix another search FP
8. Removes some duplicate rules (and also fixes another FP)
9. Fix FP when setting up a new Nextcloud Server

### CODE REVIEW

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 1]
```diff
@@ -275,27 +268,23 @@ SecRule REQUEST_FILENAME "@endsWith /apps/text/public/session/push" \
...
     t:none,\
     nolog,\
-    ctl:requestBodyProcessor=XML,\
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
-    setvar:'tx.allowed_request_content_type=%{tx.allowed_request_content_type} |text/plain|'"
+    chain"
+    SecRule REQUEST_HEADERS:Content-Type "@rx ^text/plain(?:\;charset=UTF-8)?$" \
+        "t:none,\
+        chain"
+        SecRule REQUEST_METHOD "@streq PROPFIND" \
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:284]: Does this check also work for older versions?

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 2]
```diff
@@ -720,6 +707,31 @@ SecRule REQUEST_FILENAME "@endsWith /settings/users" \
...
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetById=942450;REQUEST_COOKIES:oc_sessionPassphrase,\
+    ctl:ruleRemoveTargetById=942210;REQUEST_COOKIES:oc_sessionPassphrase,\
+    ctl:ruleRemoveTargetById=942450;REQUEST_COOKIES:nc_session_id,\
+    ctl:ruleRemoveTargetById=942450;REQUEST_COOKIES:nc_token,\
+    ctl:ruleRemoveTargetById=942450;REQUEST_COOKIES:ocddjoynjqgd"
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:733]: Is this value really stable??

**Review by theseion — CHANGES_REQUESTED** [review]

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 2 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:733]: hm... it looks to be unique to each Nextcloud instance. It's a PL-2 FP so it's not a big deal but still annoying.

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 1 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:284]: Yes, the FP was introduced in Nextcloud 27 Hub 5 (Previous release) and it works there.
I encountered another FP while double checking this and added another commit for it

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
Thanks for also taking care of clean up!

**Pull request merged**